app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksFragment.kt

```diff
@@ -154,13 +154,28 @@ class TasksFragment : Fragment() {
     }
 
     private fun setupListAdapter() {
-        val viewModel = viewDataBinding.viewmodel
-        if (viewModel != null) {
-            listAdapter = TasksAdapter(ArrayList(0), viewModel)
-            viewDataBinding.tasksList.adapter = listAdapter
-        } else {
-            Timber.w("ViewModel not initialized when attempting to set up adapter.")
+
+        val userActionsListener = object : TaskItemUserActionsListener {
+            override fun onCompleteChanged(task: Task, v: View) {
+                val checked = (v as CheckBox).isChecked
+                viewModel.completeTask(task, checked)
+
+                // TODO need some way to trigger tasks to refresh when checkbox state changed
+                viewModel.loadTasks(false)
```

Comment:
This is a STOPSHIP

Also, should not be necessary. The checkbox is shown as checked when the user checks it, doesn't it?